src/components/Terminal/Terminal.tsx

```diff
@@ -95,7 +95,17 @@ class Terminal extends React.Component<TerminalProps, {}> {
   };
 
   componentDidMount() {
-    this.props.rootStore.start();
+    if (this.props.rootStore.authStore.isAuth) {
+      this.props.rootStore.balanceListStore.fetchAll().then(() => {
+        if (this.props.rootStore.authStore.noKycAndFunds) {
+          this.props.rootStore.authStore.showNoFundsAndKycModal();
+        } else {
+          this.props.rootStore.start();
+        }
+      });
+    } else {
+      this.props.rootStore.start();
```

Comment:
we can improve here, this should call `startPublicMode()` and then we can remove  implicit call for `startPublicMode` in the `tootStore#start`